ZitiUpdateService/UpdateService.cs

```diff
@@ -811,17 +811,27 @@ public partial class UpdateService : ServiceBase {
 			Logger.Debug("downloaded file hash was correct. update can continue.");
 #if !SKIPUPDATE
 			Logger.Info("verifying file [{}]", fileDestination);
-			new SignedFileValidator(fileDestination).Verify();
-			Logger.Info("SignedFileValidator complete");
 			try {
-				StopZiti();
-				StopUI().Wait();
-
-				Logger.Info("Running update package: " + fileDestination);
-				// shell out to a new process and run the uninstall, reinstall steps which SHOULD stop this current process as well
-				Process.Start(fileDestination, "/passive");
-			} catch (Exception ex) {
-				Logger.Error(ex, "Unexpected error during installation");
+				new SignedFileValidator(fileDestination).Verify();
+				Logger.Info("SignedFileValidator complete");
+				try {
+					StopZiti();
+					StopUI().Wait();
+
+					Logger.Info("Running update package: " + fileDestination);
+					// shell out to a new process and run the uninstall, reinstall steps which SHOULD stop this current process as well
+					Process.Start(fileDestination, "/passive");
+				} catch (Exception ex) {
+					Logger.Error(ex, "Unexpected error during installation");
+				}
+			} catch (Exception e) {
+				MonitorServiceStatusEvent status = new MonitorServiceStatusEvent() {
+					Code = 2,
+					Error = "Upgrade Error, Check Monitor Logs For Details",
+					Message = ""
+				};
+				Logger.Info(e.Message);

```

Comment:
change this to "notifying the UI of the upgrade failure" and then change the catch to just `} catch {` (no exception name)